adoc/chapters/device_compiler.adoc

```diff
@@ -129,96 +136,98 @@ void h() {
 }
 ----
 
-In order for the SYCL device compiler to correctly compile <<device function, device functions>>, all
-functions in the source file, whether <<device function, device functions>> or not, must be
-syntactically correct functions according to this specification. A syntactically
-correct function adheres to at least the minimum required {cpp} version
-defined in <<sec:progmodel.minimumcppversion>>.
+In order for the SYCL device compiler to correctly compile <<device
```

Comment:
This took me a while to track down ...

Asciidoctor does not allow you to split a document reference across two lines like this.  Therefore, you should break the line before the "<<device".  As it is now, this causes the following warning when you build the spec:

```
INFO: possible invalid reference: device
```

Reply:
In the specs I am editor for, we use the convention that anchor names are hyphenated, e.g. \<\<kernel-bundle, kernel bundles>>.

Reply:
Good catch! I should have try to build it myself. I did a visual review, was obviously not enough. 

My 2cents: It may be easier for us to change the SYCL spec to use hyphen (the regex should be relatively pretty easy) than to try to patch `reflow.py`. At least for me :)

Reply:
For this type of thing, you can also do 'wdiff -3' to compare tokens in the HTML outputs with usually good results.

FWIW,

> git grep -E '<<[[:alpha:]:]+ '

turns up only these xrefs:

* 'command group'
* 'control flow'
* 'device copyable'
* 'device function'
* 'kernel bundle'

and AFAICT all of them are glossary terms which already have a corresponding hyphenated anchor.

Reply:
And we also use `<<device-function, Device functions>>`, so I guess it was so incoherence in our spec. Will fix.

Reply:
Yes, I agree that we should change these references to use the (hyphenated) anchor name.  I did this in #482.

I'm surprised that Asciidoctor doesn't issue an error when there is a reference that does not have a matching anchor.  I guess the syntax "\<\<foo\>\>" will link to *either* an anchor "[[foo']]" or to a glossary term "foo"?

Reply:
You get anchors implicitly generated in some circumstances, for example section titles (and probably glossary entries, too, though not sure about that). I prefer having explicit control over this so I use explicit anchors everywhere.

There are varying levels of warning / error behavior selectable through command line options. They probably are not turned all the way up at present.

Reply:
BTW I think if you do something like '\<\<device
anchor, some text>>' you get a generated '\<a href="device
anchor">' which I guess? is legal HTML? But does not map to 'device-anchor'.